tuxemon/save_upgrader.py

```diff
@@ -78,6 +78,12 @@ def upgrade_save(save_data: Dict[str, Any]) -> Dict[str, Any]:
     if "steps" not in save_data["game_variables"]:
         save_data["game_variables"]["steps"] = 0
 
+    first_steps = [('taba_town','xero_taba_town'),('player_house_bedroom','xero_player_house_bedroom'),('route1','xero_route1'),
+    ('player_house_downstairs','xero_player_house_downstairs'),('route1_sanglorian','xero_route1_sanglorian'),
+    ('tuxe_mart_taba','xero_tuxe_mart_taba'),('healing_center','xero_healing_center'),('professor_lab','xero_professor_lab')]
+    for k, v in first_steps:
```

Comment:
This seems dangerous.

route1_sanglorian -> xero_route1_sanglorian -> xero_xero_route1_sanglorian

Just make the original data structure a dict and do a single lookup of the current value. No looping.

If somebody ever reuses one of the current names then it will be converted. Bump the version number and only run this logic for old saves

Reply:
Run black on the code.